jbpm-workitems/jbpm-workitems-email/src/test/java/org/jbpm/process/workitem/email/EmailNotificationPublisherTest.java

```diff
@@ -0,0 +1,177 @@
+/*
+ * Copyright 2017 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.process.workitem.email;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNull;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.Map;
+import java.util.Set;
+
+import javax.mail.internet.InternetAddress;
+import javax.mail.internet.MimeMessage;
+import javax.mail.internet.MimeMessage.RecipientType;
+
+import org.jbpm.services.task.identity.DefaultUserInfo;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+import org.kie.api.task.model.OrganizationalEntity;
+import org.kie.internal.task.api.TaskModelFactory;
+import org.kie.internal.task.api.TaskModelProvider;
+import org.kie.internal.task.api.UserInfo;
+import org.kie.internal.utils.ChainedProperties;
+import org.kie.internal.utils.ClassLoaderUtil;
+import org.subethamail.wiser.Wiser;
+import org.subethamail.wiser.WiserMessage;
+
+public class EmailNotificationPublisherTest {
+
+    private Wiser wiser;
+
+    private String emailHost;
+    private String emailPort;
+    
+    private Connection connection;
+    private UserInfo userInfo;
+    
+    private TaskModelFactory factory = TaskModelProvider.getFactory();
+
+    @Before
+    public void setUp() throws Exception {
+        System.setProperty("org.jbpm.email.templates.dir", new File("src/test/resources/templates").getAbsolutePath());
+        TemplateManager.reset();
+        
+        ChainedProperties props = ChainedProperties.getChainedProperties( "email.conf", ClassLoaderUtil.getClassLoader( null, getClass(), false ));
+        emailHost = props.getProperty( "mail.smtp.host", "localhost" );
+        emailPort = props.getProperty( "mail.smtp.port", "2345" );
+
+        wiser = new Wiser();
```

Comment:
Is it really necessary to set up the Wiser before each test method? My guess would be that `@BeforeClass` should be fine.

Reply:
I want to always use a clean set as in few situations class level wiser caused some issues...

Reply:
if there were some issues with wiser, +1